Core/include/Acts/TrackFinding/CombinatorialKalmanFilter.hpp

```diff
@@ -927,23 +926,27 @@ class CombinatorialKalmanFilter {
     /// @param gctx The geometry context for this track finding/fitting
     /// @param newTrackStateList index list of new track states
     /// @param result which contains among others the new states, and the list of active branches
-    /// @return the number of newly added branches and number of stopped branches or an error
-    Result<std::pair<unsigned int, unsigned int>> processNewTrackStates(
+    /// @return the number of newly added branches or an error
+    Result<unsigned int> processNewTrackStates(
         const Acts::GeometryContext& gctx,
         const CkfTypes::BranchVector<TrackIndexType>& newTrackStateList,
         result_type& result) const {
       using PM = TrackStatePropMask;
 
       unsigned int nBranchesOnSurface = 0;
-      unsigned int nStoppedBranches = 0;
 
       auto rootBranch = result.activeBranches.back();
 
-      // Build the new branches by forking the root branch
+      // Build the new branches by forking the root branch. Reverse the order to
+      // process the best candidate first
       CkfTypes::BranchVector<TrackProxy> newBranches;
-      for (auto [i, tipIndex] : enumerate(newTrackStateList)) {
```

Comment:
Not quite sure I understand the reason of this change. 
- the assembly created by the two version should bemostly identical 
- newTrackStateList should be sorted by chi2 (increasing order)

Reply:
reason being that we store the chi2 in increasing order but the branches are processed in the inverse order so back first. to process the best branch first I reversed the order of the measurements

Reply:
Just to answer myself, if I am not mistaken the new branches are "popped" from the top. Thus, adding the "best" state last would lead to the corresponding branch to be processed first.